Lib/glyphsLib/writer.py

```diff
@@ -166,3 +164,64 @@ def dumps(obj):
     fp = UnicodeIO()
     dump(obj, fp)
     return fp.getvalue()
+
+
+NSPropertyListNameSet = (
+    # 0
+    False, False, False, False, False, False, False, False,
+    False, False, False, False, False, False, False, False,
+    # 16
+    False, False, False, False, False, False, False, False,
+    False, False, False, False, False, False, False, False,
+    # 32
+    False, False, False, False, True, False, False, False,
+    False, False, False, False, False, False, True, False,
+    # 48
+    True, True, True, True, True, True, True, True,
+    True, True, False, False, False, False, False, False,
+    # 64
+    False, True, True, True, True, True, True, True,
+    True, True, True, True, True, True, True, True,
+    # 80
+    True, True, True, True, True, True, True, True,
+    True, True, True, False, False, False, False, True,
+    # 96
+    False, True, True, True, True, True, True, True,
+    True, True, True, True, True, True, True, True,
+    # 112
+    True, True, True, True, True, True, True, True,
+    True, True, True, False, False, False, False, False
+    )
+
+
+def _needs_quotes(string):
+    if len(string) == 0:
+        return True
+    if not isinstance(string, (str, unicode)):
+        return False
+
+    # Does it need quotes because of special characters?
+    for c in string:
+        d = ord(c)
+        if d >= 128 or not NSPropertyListNameSet[d]:
+            return True
+
+    # Does it need quotes because it could be confused with a number?
+    i = None
+    try:
+        i = int(string)
+    except:
+        pass
+    if i is not None:
+        return True
+
+    return False
+
+
+def escape_string(value):
+    if isinstance(value, (str, unicode)) and _needs_quotes(value):
+        value = value.replace("\\", "\\\\")
+        value = value.replace("\"", "\\\"")
+        value = value.replace("\n", "\\012")
+        value = '"%s"' % value
```

Comment:
same thing here, maybe re.sub with a replace callback is better than multiple replace.
it's a pity that the signature of the `translate` method of built-in strings changed between python2 and 3, because that would have been handy in these situations.